builder/amazon/common/block_device.go

```diff
@@ -139,11 +145,16 @@ func (blockDevice BlockDevice) BuildEC2BlockDeviceMapping() *ec2.BlockDeviceMapp
 		ebsBlockDevice.VolumeSize = aws.Int64(blockDevice.VolumeSize)
 	}
 
-	// IOPS is only valid for io1 and io2 types
-	if blockDevice.VolumeType == "io1" || blockDevice.VolumeType == "io2" {
+	switch blockDevice.VolumeType {
+	case "io1", "io2", "gp3":
 		ebsBlockDevice.Iops = aws.Int64(blockDevice.IOPS)
 	}
 
+	// Throughput is only valid for gp3 types
```

Comment:
Do you think it would be a good idea to error here when `Throughput` is set for something else than gp3 ?

Reply:
yeah, I can do that. I was just following the pattern from iops where it only used it when appropriate. I'll patch it shortly